include/dt-bindings/pinctrl/stm32-pinctrl-common.h

```diff
@@ -7,6 +7,31 @@
 #ifndef ZEPHYR_STM32_PINCTRL_COMMON_H_
 #define ZEPHYR_STM32_PINCTRL_COMMON_H_
 
+/* Extracted from Linux: include/dt-bindings/pinctrl/stm32-pinfunc.h */
+/*  define PIN modes */
+#define GPIO	0x0
+#define AF0	0x1
```

Comment:
I would start the numbering with AF0 0x0 to use the alternate functions directly for controlling the hardware.

In case you use something like this in the pinctrl/pinmux driver to get the LL_GPIO_ALT_FUNC numbers:
```
static inline u32_t pinctrl_stm32_ll_gpio_alt_func(u16_t func)
{
	if ((func >= AF0) &&
	    (func <= AF15)) {
		return func - AF0;
	}
	__ASSERT(0, "function 0x%x unknown - no LL_GPIO_ALT_FUNC", (int)func);
	return func;
}
```
the compiler can eliminate at least the >= AF0 check.